### PR TITLE
Allow user-selectable brand image

### PR DESCRIPTION
**Pull request by jaxxzer**
Some of our users use screen grab programs to produce deliverables and would like to have their business identified in the shot. This PR allows for that.

screenshot 2017-09-14 12 04 52

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
"Link Brand Images"?

**Comment by jaxxzer** [comment]
@DonLakeFlyer 'Link Brand Images' = Use the same image for both indoor and outdoor theme.

**Comment by DonLakeFlyer** [comment]
Seems unnecessary. Simple better?

**Comment by jaxxzer** [comment]
Ok, maybe remove the 'Link' option, and use the default image if the user has not selected a custom image for the current theme.

Or just one image field that will be used for both themes?

**Comment by DonLakeFlyer** [comment]
Use the opposite themed brand image if they only specify one.

**Comment by jaxxzer** [comment]
Got it.

**Comment by DonLakeFlyer** [comment]
Could you add doc in the Daily Build new features page of the qgc docs for this?

**Comment by jaxxzer** [comment]
Yes.

**Comment by DonLakeFlyer** [comment]
Also the whole section group box and the individual fields need to respect SettingsFact.visible state. This allows custom builds to hide things.

**Comment by jaxxzer** [comment]
Ooo, thanks for the tip :+1:

**Comment by dheideman** [comment]
Brand image settings without "link" fact:

removed linked option

Still need to fix the "visible" state.

**Comment by dheideman** [comment]
I just realized that all the brand image facts are lumped in with the AppSettings group internally but have a separate box on the settings page.  Would it be better to move the image path selection into the Miscellaneous box in General Settings, move the facts into their own group internally, or leave it as-is?

File: src/ui/toolbar/MainToolBarIndicators.qml [hunk 1]
```diff
@@ -100,8 +100,8 @@ Item {
...
         property bool   _userBrandingIndoor:    _userBrandImageIndoor.length != 0
         property bool   _userBrandingOutdoor:   _userBrandImageOutdoor.length != 0
-        property string _brandImageIndoor:      _userBrandingIndoor ? _userBrandImageIndoor : (_corePluginBranding ? QGroundControl.corePlugin.brandImageIndoor : (_activeVehicle ? _activeVehicle.brandImageIndoor : "") )
-        property string _brandImageOutdoor:     _userBrandingOutdoor ? _userBrandImageOutdoor : (_corePluginBranding ? QGroundControl.corePlugin.brandImageOutdoor : (_activeVehicle ? _activeVehicle.brandImageOutdoor : "") )
+        property string _brandImageIndoor:      _userBrandingIndoor ? _userBrandImageIndoor : (_userBrandingOutdoor ? _userBrandImageOutdoor : (_corePluginBranding ? QGroundControl.corePlugin.brandImageIndoor : (_activeVehicle ? _activeVehicle.brandImageIndoor : "") ) )
+        property string _brandImageOutdoor:     _userBrandingOutdoor ? _userBrandImageOutdoor : (_userBrandingIndoor ? _userBrandImageIndoor : (_corePluginBranding ? QGroundControl.corePlugin.brandImageOutdoor : (_activeVehicle ? _activeVehicle.brandImageOutdoor : "") ) )
```
Comment (jaxxzer) [inline src/ui/toolbar/MainToolBarIndicators.qml]: @dheideman Can you reformat this for readability?

**Comment by dheideman** [comment]
@DonLakeFlyer I went ahead and made a new fact group for Brand Image since it doesn't feel like it would fit in with Miscellaneous.  Let me know if there's a better way to handle this.

**Comment by DonLakeFlyer** [comment]
I think that's fine. That way easier to hide the whole thing from the FactGroup

**Pull request merged**